constrain-lib/src/main/java/org/gradle/dependency/constrain/lib/serialize/ConstraintsJsonReader.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.dependency.constrain.lib.serialize;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaException;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import org.gradle.dependency.constrain.lib.DependencyConstrainException;
+import org.gradle.dependency.constrain.lib.model.LoadedConstraint;
+import org.gradle.dependency.constrain.lib.model.LoadedConstraints;
+
+import javax.annotation.WillClose;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Set;
+
+public class ConstraintsJsonReader {
+    private static final String CONSTRAINTS_SPECIFICATION = "schema/dependency-constraints-schema.json";
+
+    public static LoadedConstraints readFromJson(@WillClose InputStream in) {
+        JsonSchema schema = createSchemaValidator();
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
`ObjectMapper` is threadsafe and can be shared, so having a static instance might be better.

Reply:
I want it garbage collected when I'm done here. I did think about that.

Reply:
What is the usage scenario of this class then, if it is one off execution, then it shouldn't matter that it is a static field. If it is used multiple times during a longer running execution then caching the object mapper would also make sense.

Reply:
The usage scenario for this class is it will be invoked the same number of times per build as the Gradle Verification Metadata is loaded during a build execution. As such, it should only happen a few times per build. Ideally it would only happen once, but that's not currently possible given the current Gradle DI infrastructure setup with respect to how `buildSrc` and included builds are invoked.